src/main/java/io/inisos/bank4j/util/Iso20022ReferenceElementValidator.java

```diff
@@ -0,0 +1,71 @@
+package io.inisos.bank4j.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class Iso20022ReferenceElementValidator {
+    private static final Pattern ISO20022_VALID_CHARS_PATTERN = Pattern.compile("^[A-Za-z0-9\\(\\)\\+\\,\\-\\./:\\s]+$");
+    private static final Pattern ISO20022_INVALID_START_OR_END_PATTERN = Pattern.compile("^[/\\s]|[//\\s]$");
```

Comment:
SonarLint:
S5850: Group parts of the regex together to make the intended operator precedence explicit.

Reply:
Replaced by `^(?:[/\\s]|[//\\s])$`